src/sponsorableContributorsFetcher.ts

```diff
@@ -0,0 +1,114 @@
+import { Config } from './config';
+import { ContributionCountOfUserIntoRepo, RepositoryName } from './contributorFetcher';
+import { graphql } from '@octokit/graphql';
+import { GraphQlQueryResponseData, RequestParameters } from '@octokit/graphql/dist-types/types';
+
+export interface Sponsor {
+    __typename: string;
+    email?: string;
+    login: string;
+    url: string;
+    sponsorsListing: {
+        name: string | null;
+        dashboard: string | null;
+    };
+}
+
+// query WIP: fetchiing first 10 sponsors, need to add pagination
+export const queryTemplate = `query fetchSponsorable($queryString: String!) {
+  search(query: $queryString, type: USER, first: 10) {
+    nodes {
+      ... on User {
+        __typename
+        email
+        login
+        url
+        sponsorsListing {
+          name
+          dashboardUrl
+        }
+      }
+        ... on Organization {
+        __typename
+        login
+        url
+        sponsorsListing {
+          name
+          dashboardUrl
+        }
+      }
+    }
+  }
+}`;
+
+interface Variables extends RequestParameters {
+    queryString: string;
+}
+
+export class SponsorableContributorsFetcher {
+    private readonly config: Config;
+
+    public constructor(config: Config) {
+        this.config = config;
+    }
+
+    public async fetchSponsorableContributorsInformation(
+        token: string,
+        query: string,
+        contributors: Map<RepositoryName, ContributionCountOfUserIntoRepo[]>
+    ): Promise<Map<string, Sponsor[]>> {
+        const allSponsorableUsersInfo = [];
+        const sponsorables = new Map<string, Sponsor[]>();
+
+        for (const [repoIdentifier, githubUsers] of contributors.entries()) {
```

Comment:
To avoid dealing with "entries", I would typically have this outer loop iterate through just the "keys". Then the first line inside the loop would be `const users = contributors.get(repoIdentifier);`